controllers/status.go

```diff
@@ -62,7 +62,7 @@ func (r *GslbReconciler) getServiceHealthStatus(gslb *k8gbv1beta1.Gslb) (map[str
 			service := &corev1.Service{}
 			finder := client.ObjectKey{
 				Namespace: gslb.Namespace,
-				Name:      path.Backend.ServiceName,
+				Name:      path.Backend.Service.Name,
```

Comment:
right, backend itself can't be nil because it doesn't have the "omitempty" tag, but `path.Backend.Service` can be nil (it's `path.Service` XOR `path.Resource` we support only the Service ref)

If service is there, then the Name has to be specified (again not "omniempty" tag). But to stay on the safe side, I can be paranoid and check everything :)

Reply:
I've added some checks and logged the warning msg if it's malformed. Also marking the svc health as `k8gbv1beta1.NotFound`... in a sense: "the info user gave us was not sufficient to find the service"

Hopefully, it won't be that spammy in the logs. It's going to be there in each reconciliation, but on the other hand, if we "swallowed" the error, it's even worse.

Reply:
ok, lets keep it as it is and we will see afterwards. Possible candidate for validation would be depresolver_spec.